source/common/upstream/load_balancer_impl.h

```diff
@@ -40,24 +40,7 @@ class LoadBalancerBase : public LoadBalancer {
   choosePriority(uint64_t hash, const HealthyLoad& healthy_per_priority_load,
                  const DegradedLoad& degraded_per_priority_load);
 
-  HostConstSharedPtr chooseHost(LoadBalancerContext* context) override;
```

Comment:
Just so I understand, why did you move this down in the class hierarchy? Thanks

Reply:
This question is a bit more complicated.

Both `ThreadAwareLoadBalancerBase` and `ZoneAwareLoadBalancerBase` need a cross priority host map. But the use of the two is completely different. The former needs to ensure that the host map is thread-safe, while the latter does not need to consider this issue at all.

This led me to finally decide to add a member of cross_priority_host_map_ to `ThreadAwareLoadBalancerBase` and `ZoneAwareLoadBalancerBase` separately, instead of adding a member of cross_priority_host_map_ to the common base class `LoadBalancerBase` of the two.

But this brings a new problem, that is, the `chooseHost` method in `LoadBalancerBase` needs to access cross_priority_host_map_ to search for override host.
Although I know that the base class can access the members of the derived class through virtual functions, but there is no need to be so complicated.

The `chooseHost` method in `LoadBalancerBase` is only meaningful to `ZoneAwareLoadBalancerBase` and its derived classes, so there is no need to implement it in `LoadBalancerBase`.

So, I finally moved `chooseHost` to `ZoneAwareLoadBalancerBase`, so that it will not affect the function, but also can directly access cross_priority_host_map_ in `chooseHost`, and it is simpler and cleaner.

Reply:
Thanks for explaining! This makes sense to me